transform/binary-plugin/binary_plugin.go

```diff
@@ -123,12 +123,20 @@ func (b *binaryRunner) Metadata(log logrus.FieldLogger) ([]byte, []byte, error)
 
 }
 
-func (b *binaryRunner) Run(u *unstructured.Unstructured, log logrus.FieldLogger) ([]byte, []byte, error) {
+func (b *binaryRunner) Run(u *unstructured.Unstructured, extras map[string]string, log logrus.FieldLogger) ([]byte, []byte, error) {
 	objJson, err := u.MarshalJSON()
 	if err != nil {
 		log.Errorf("unable to marshal unstructured Object")
 		return nil, nil, fmt.Errorf("unable to marshal unstructured Object: %s, err: %v", u, err)
 	}
+	if len(extras) > 0 {
+		extrasJson, err := json.Marshal(extras)
+		if err != nil {
+			log.Errorf("unable to marshal extras map")
+			return nil, nil, fmt.Errorf("unable to marshal extras map: %v, err: %v", extras, err)
+		}
+		objJson = append(objJson, extrasJson...)
```

Comment:
Instead of adding a second JSON object, we should consider as a follow-up an object that encapsulates this and making it a versioned API.

Reply:
@shawn-hurley So it sounds like this isn't a merge blocker but something to follow-up with?  Maybe  something like this to replace the expectation of two separate objects:
```
type PluginRequest struct {
    Obj unstructured.Unstructured
    Extras map[string[string
    Version string
}
```